solutions/swb-app/src/accountRoutes.ts

```diff
@@ -19,11 +19,19 @@ import { escape } from 'lodash';
 import { wrapAsync } from './errorHandlers';
 import { processValidatorResult } from './validatorHelper';
 
-export function setUpAccountRoutes(router: Router, account: HostingAccountService): void {
+export function setUpAccountRoutes(router: Router, hostingAccountService: HostingAccountService): void {
+  router.get(
+    '/aws-accounts',
+    wrapAsync(async (req: Request, res: Response) => {
+      const listRequest = ListAccountsRequestParser.parse(req.params);
```

Comment:
I think we should throw a bad request instead of a 500, or  if we don't want it to be part of the scope we should add a ticket so we don't forget it in the future

Reply:
Interesting, are we moving away from using json-schema to validate requests? That's what is being used in other APIs. For example, line 41 of this file uses schema.

We might want to differentiate between the schema of the API params and the Request object for the service itself. Doing it this way means that `req.params` have to have the same attribute as the `ListAccountRequest` object. Is this a restriction we're ok with imposing on all APIs?

For example, what if we wanted to add an extra attribute to `listRequest` that is not a part of `req.params`? Like for example, we want to add a feature flag to `listRequest` before sending the request to `hostingAccountService.list`.

Note, we also have plans to move request schemas to `swb-app`. [(link)](https://quip-amazon.com/SmkEAs1xqZBy/SWBv2-Request-Object-and-Schema)

This allows developer to choose their own schema for the API. And then it's up to the developer to convert the schema to the correct request object for using the `ResourceService`.

Reply:
How I see it is that the same way json schema only validates the request object , zod will only validate the request object, and then we are sure the object has the format we need so we can just parse the request into the service object if its needed, in this case there was nothing needed but at least the service doesn't have to worry about the format of the object, it will only validate values not formats

Reply:
I'm in favor of moving to zod to get additional functionality, while still achieving the same goals that json schema achieves. The goal of a validator for requests is to ensure that all req.params match the expected request object anyways, so I don't see the need for handling extra attributes that aren't clearly defined in the request parser.

Reply:
I'm ok with using zod for validating api requests and service request. However, the api request schema and service request schema shouldn't be tightly coupled. Another solution might want to use our services but might want different values for the api request parameters. For the example we might use project as part of our api param but DEA might want to use group as the api param instead.